api/bucketsd/service.go

```diff
@@ -970,16 +972,143 @@ func (s *Service) SetPath(ctx context.Context, req *pb.SetPathRequest) (res *pb.
 	}
 	dbToken, _ := thread.TokenFromContext(ctx)
 
-	destPath := cleanPath(req.Path)
-	bootCid, err := cid.Decode(req.Cid)
+	cleanPth, err := parsePath(req.FromPath)
 	if err != nil {
-		return nil, fmt.Errorf("invalid remote cid: %v", err)
+		return nil, err
+	}
+	cleanDst, err := parsePath(req.ToPath)
+	if err != nil {
+		return nil, err
+	}
+
+	if cleanPth == cleanDst {
+		return &pb.MovePathResponse{}, nil
 	}
 
 	lck := s.Semaphores.Get(buckLock(req.Key))
 	lck.Acquire()
 	defer lck.Release()
 
+	_, pth, err := s.getBucketPath(ctx, dbID, req.Key, cleanPth, dbToken)
+	if err != nil {
+		return nil, err
+	}
+	n, err := s.getNodeAtPath(ctx, pth, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = s.setPath(ctx, &pb.SetPathRequest{
+		Key:  req.Key,
+		Path: cleanDst,
+		Cid:  n.Cid().String(),
+	}, dbID, dbToken, cleanDst)
```

Comment:
there is no way this is going to work on an encrypted bucket, right @sanderpick ? the source CID is for the encrypted dag... 

it will fail down at line 1029

the `err` is `could not resolve path: /ipfs/bafybeieq2ts26nrhfldfdkkafdqssocijpdtie746nwvih54m6zdzy2cya/a/.textileseed`.

if i pop a listPath of the destination in at line 1029 before returning, i see

```
item:{cid:"bafybeiczsscdsbs7ffqz55asqdf3smv6klcw3gofszvwlyarci47bgf354" name:"a" path:"/ipfs/bafybeib3l6pbh55k6iparpsebbkmkabzdxtt5vkq3pempnlgsvqpghta7y/a" size:4 is_dir:true metadata:{updated_at:1613596140630032000}} root:{key:"bafzbeiend6uzkwh2dadgl3h6v7smnmlqp2hxkcn7vcwpy4ekaaibuqmf54" path:"/ipfs/bafybeib3l6pbh55k6iparpsebbkmkabzdxtt5vkq3pempnlgsvqpghta7y" created_at:1613596140341415000 updated_at:1613596140630032000 thread:"bafk26pxojma4xalp4x2sqhnmgzuhaa6phywo4qlpwkwxfdmo4ctofzi" owner:"bbaareic6kv6huvr7fbhdajm2tv42iriybrzqo6exi3iylsw7bwthv4az7u" version:1 metadata:{roles:{key:"bbaareic6kv6huvr7fbhdajm2tv42iriybrzqo6exi3iylsw7bwthv4az7u" value:PATH_ACCESS_ROLE_ADMIN} updated_at:1613596140341415000 key:"biq/HrgkeftcYQaWlc52R8guyCgBrBOeGmIc3uJGeGgRgK9AayFrN3L1aYHcxquqXgZiDqU0GnQKeHBgIkSTfg=="} link_key:"5RYdG/U3nhk/Vt7gy1Yq7qAWL0gbmkTNO3Ht/K4QZErPp1L54vPIuzvyjnNJIMmKHapfUa98UEygom5GKoj9Gg==" path_metadata:{key:"" value:{roles:{key:"bbaareic6kv6huvr7fbhdajm2tv42iriybrzqo6exi3iylsw7bwthv4az7u" value:PATH_ACCESS_ROLE_ADMIN} updated_at:1613596140341415000 key:"biq/HrgkeftcYQaWlc52R8guyCgBrBOeGmIc3uJGeGgRgK9AayFrN3L1aYHcxquqXgZiDqU0GnQKeHBgIkSTfg=="}} path_metadata:{key:".textileseed" value:{updated_at:1613596140341415000}} path_metadata:{key:"a" value:{updated_at:1613596140630032000}} path_metadata:{key:"file1.jpg" value:{updated_at:1613596140452829000}} path_metadata:{key:"file2.jpg" value:{updated_at:1613596140463238000}} archives:{current:{}}}
```

so i can see it there in the metadata... but i dunnooo.... maybe result at that node is actually going to be doubly encrypted (original encryption, plus setpath adds another encryption...) not sure. will start walking through the encrypted flow more closely now

Reply:
I'll take a look in a bit... I think you'll have to use a more internal method (or slightly modified one) to _add the node at the path_, not the public API version (`SetPath`).

Reply:
Since the node won't be available from the block store, but it _is_ in memery.